### PR TITLE
Fix one source that gives the leap year simulation day missing problem.

### LINKED ISSUE #8772: Missing simulation days when starting with a leap year and end with the next year
Issue overview
--------------
It was reported that there are missing simulation days when dealing with leap starting year. For a full year simulation (expecting 365 days), one to two days of simulations were found missing at the end of the running period. 

After some experiment runs, one common problematic scenario is when the simulation start year is a leap year, but the end year is next year. In this case, the number of total simulation days would get a 364 instead of 365 if a full year is specified—for example, running from Oct 1, 2020 to Sep 30, 2021 will only get simulation runs (and reports) s from Oct 1, 2020 to Sep 29, 2021, with the last day (Sep 30, 2021) missing. 

Actually this would happen to any ordinary idf file---just ordinary year and no special specification for "leap year weather" or  "actual weather treatment". See the attached idf file modified from the example file 1ZoneUncontrolled_DD2009.idf. The problem will occur when the simulation period from Oct 1, 2016 to Sep 30, 2021 in any file with default setting with typical default (blank) and typical weather file (default Atlantic weather with `HOLIDAYS/DAYLIGHT SAVINGS,No,0,0,0`--so it is just a case for ordinary year and not for leap year at all, but the problem happens and the simulation only run from 2016-10-01 to 2017-09-29.

The root cause is the processing of the start Julian day inputs have some missing logic in handling the case of having a starting leap year and then a different ending year. 

[1ZoneUncontrolled_DD2009_rev1.idf.txt](https://github.com/NREL/EnergyPlus/files/6509220/1ZoneUncontrolled_DD2009_rev1.idf.txt)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus (if using an intermediate build, include SHA): v9.4 v9.5
 - Refer to Helpdesk ticket number 15720

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Tries to fix #8772
 - The very general case was fixed. Some further fix might needed and tested for some other corner cases and combinations with the weather file settings.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/WeatherManager.cc [hunk 1]
```diff
@@ -9117,7 +9117,8 @@ namespace WeatherManager {
                 } else {
                     // Environment crosses year boundaries
                     env.RollDayTypeOnRepeat = runPer.RollDayTypeOnRepeat;
-                    env.StartJDay = General::OrdinalDay(runPer.startMonth, runPer.startDay, runPer.isLeapYear ? 1 : 0);
+                    env.StartJDay = General::OrdinalDay(
+                        runPer.startMonth, runPer.startDay, isLeapYear(env.StartYear) && state.dataWeatherManager->WFAllowsLeapYears ? 1 : 0);
```
Comment (jasondegraw) [inline src/EnergyPlus/WeatherManager.cc:9202]: This change means that `env.IsLeapYear`/`runPer.isLeapYear` aren't definitive on whether the first year of the environment is a leap year or not, so this is not going to work by itself. The reported issue needs more information before it's at all clear what the fix is.

**Comment by Myoldmopar** [comment]
Because of the potential impact of this, should we try to get it in early for IO Freeze, or punt until after 9.6?  I don't want something potentially volatile to squeeze into bug fix release time.

**Comment by jcyuan2020** [comment]
The complete fix could be more complex. The major function related to this, is functioning fine if the user take a little bit caution. So I think this might not be that urgent and could be left to future development after v9.6. 

BTW, the user also reported another corner case abnormality issue, which I will post so someone could also take a look in the future if needed. Again, the major functionality related to this leap year is functioning fine if the users take some cautions to avoid going to these rare corner case `traps`.

**Comment by Myoldmopar** [comment]
Excellent, let's push this to after 9.6 then.  Thanks for the full update!

**Comment by mjwitte** [comment]
Just a one-liner here, so closing this PR. Will open a new on when work resumes, post v9.6.

**Pull request closed without merging**